MapboxNavigation/RouteMapViewController.swift

```diff
@@ -684,6 +684,15 @@ extension RouteMapViewController: NavigationViewDelegate {
             streetLabelLayer.lineOpacity = NSExpression(forConstantValue: 1)
             streetLabelLayer.lineWidth = NSExpression(forConstantValue: 20)
             streetLabelLayer.lineColor = NSExpression(forConstantValue: UIColor.white)
+
+            if ![DirectionsProfileIdentifier.walking, DirectionsProfileIdentifier.cycling].contains( router.routeProgress.routeOptions.profileIdentifier) {
+                // filter out to road classes valid for motor transport
+                let roadPredicates = ["motorway", "motorway_link", "trunk", "trunk_link", "primary", "primary_link", "secondary", "secondary_link", "tertiary", "tertiary_link", "street", "street_limited", "roundabout", "mini_roundabout"].compactMap { roadClass -> NSPredicate? in
+                    return NSPredicate(format: "%K == %@", "class", roadClass)
+                }
+                let compoundPredicate = NSCompoundPredicate(orPredicateWithSubpredicates: roadPredicates)
```

Comment:
Something along these lines should work and would probably perform slightly better:

```suggestion
                let roadPredicates = ["motorway", "motorway_link", "trunk", "trunk_link", "primary", "primary_link", "secondary", "secondary_link", "tertiary", "tertiary_link", "street", "street_limited", "roundabout", "mini_roundabout"]
                let compoundPredicate = NSPredicate(format: "class IN %@", roadPredicates)
```

Reply:
I'll adopt these changes. Worth getting the best performance.